scripts/deploy-production.js

```diff
@@ -0,0 +1,126 @@
+const prompts = require("prompts");
+const { exec } = require("child_process");
+
+const localDeployScript =
+  "yarn build:production && env-cmd -e production yarn deploy:do-not-use && env-cmd -e production ./scripts/email.sh";
+
+const main = async () => {
+  if (!(await isOnMainBranch())) {
+    console.log("You must be on the main branch to deploy!");
+    return;
+  }
+  if (!(await isWorkingDirectoryClean())) {
+    console.log("You must have a clean working directory to deploy");
+    return;
+  }
+
+  // Print all commits between the last tag and the current commit
+  const commitMessages = await getCommitMessages();
+  let shouldDeployLocal = false;
+  if (commitMessages.length > 0) {
+    console.log("Commit messages:");
+    console.log(commitMessages);
+  } else {
+    // If there are no commit messages, ask the user if they want to deploy anyway
+    const response = await prompts({
+      type: "confirm",
+      name: "value",
+      message: "No new commits. Do you want to deploy anyway?",
+      initial: false,
+    });
+    shouldDeployLocal = response.value;
+    if (!shouldDeployLocal) {
+      return;
+    }
+  }
+
```

Comment:
It might be slightly more readable if we format this part with just `if` statements rather than an `if-else` block. Something like this?

```javascript
  const commitMessages = await getCommitMessages();
  let shouldDeployLocal = false;

  if (commitMessages.length === 0) {
    const response = await prompts({
      type: "confirm",
      name: "value",
      message: "No new commits. Do you want to deploy anyway?",
      initial: false,
    });
    shouldDeployLocal = response.value;
    if (!shouldDeployLocal) {
      return;
    }
  }

  // Print all commits between the last tag and the current commit
  if (commitMessages.length > 0) {
    console.log("Commit messages:");
    console.log(commitMessages);
  }
```